addon/-private/system/model/internal-model.js

```diff
@@ -505,6 +505,7 @@ export default class InternalModel {
     assert("Cannot destroy an internalModel while its record is materialized", !this.record || this.record.get('isDestroyed') || this.record.get('isDestroying'));
 
     this.store._removeFromIdMap(this);
+    this.store._relationshipsPayloads.unload(this.modelName, this.id);
```

Comment:
should we unify 507 and 508 into `this.store._internalModelDestroyed(internalModel)` that way, we don't need to leak `_removeFromIdMap` and `_relationshipsPayload` private members to `internalModel`